packages/flutter/lib/src/widgets/view.dart

```diff
@@ -170,24 +174,79 @@ class View extends StatelessWidget {
         ?? RendererBinding.instance.rootPipelineOwner;
   }
 
+  @override
+  State<View> createState() => _ViewState();
+}
+
+class _ViewState extends State<View> with WidgetsBindingObserver {
+  final FocusScopeNode _scopeNode = FocusScopeNode(
+    debugLabel: kReleaseMode ? null : 'View Scope',
+  );
+  final FocusTraversalPolicy _policy = ReadingOrderTraversalPolicy();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    _scopeNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didChangeViewFocus(ViewFocusEvent event) {
+    final FlutterView view = widget.view;
+    if (event.viewId != view.viewId) {
+      // The event is not pertinent to this view.
+      return;
+    }
+    FocusNode? nextFocus;
+    switch (event.state) {
+      case ViewFocusState.focused:
+        switch (event.direction) {
+          case ViewFocusDirection.forward:
+            nextFocus = _policy.findFirstFocus(_scopeNode, ignoreCurrentFocus: true);
+          case ViewFocusDirection.backward:
+            nextFocus = _policy.findLastFocus(_scopeNode, ignoreCurrentFocus: true);
+          case ViewFocusDirection.undefined:
+        }
+        (nextFocus ?? _scopeNode).requestFocus();
```

Comment:
Is this falling off the internal case to the implementation of the outer one? This syntax is mind-blowing!

Reply:
No, it just doesn't handle `ViewFocusDirection.undefined` in the inner switch, and so `nextFocus` could still be undefined at this point. If that happens, I fall back to the scope focusing the node that it last had focused.

Reply:
nit: we can get rid of the nullable value by defaulting it to _scopeNode

Either

```
FocusNode node;

switch (foo) {
 case bar: 
   node = ...;
  default:
    node = _scopeNode;
}
```

Reply:
You could add a default, but then the lint about missing cases wouldn't fire in the future if a value was added to the enum.

And if I assign it ahead of time, then there's an extra assignment being done that doesn't need to happen in most cases. (This is arguably not a big deal, though, since it's also avoiding an implicit conditional in the "??" operator.)

I think I'll just assign `nextFocus = _scopeNode` in the undefined case instead.